### PR TITLE
[sparse] Add BCSR from_scipy_sparse.

### PR DESCRIPTION
**Pull request by tlu7**
A constructor that takes scipy sparse matrix as input.

### CODE REVIEW

**Comment by tlu7** [comment]
@jakevdp

File: tests/sparse_test.py [hunk 1]
```diff
@@ -2549,6 +2549,19 @@ def test_bcoo_from_scipy_sparse(self, input_type, shape, dtype):
     M_bcoo = sparse.BCOO.from_scipy_sparse(M_sparse)
     self.assertArraysEqual(M, M_bcoo.todense())
 
+  @jtu.sample_product(
+    input_type=[scipy.sparse.coo_matrix, scipy.sparse.csr_matrix,
+                scipy.sparse.csc_matrix],
+    shape=[(5, 8), (8, 5), (5, 5), (8, 8)],
+    dtype=jtu.dtypes.floating + jtu.dtypes.complex,
+  )
+  def test_bcsr_from_scipy_sparse(self, input_type, shape, dtype):
```
Comment (tlu7) [inline tests/sparse_test.py:2558]: We could potentially merge this test with `test_bcoo_from_scipy_sparse` and call it `test_bcxx*`. Is that a good way to consolidate the tests?

Comment (jakevdp) [reply tests/sparse_test.py:2558]: Yes, I'd say consolidate the tests. You can iterate over `cls=[sparse.BCOO, sparse.BCSR]` in the test parameterization

**Review by jakevdp — APPROVED** [review]
Let's just consolidate tests, otherwise this looks great!

File: tests/sparse_test.py (hunk 1 above)
Comment (tlu7) [reply tests/sparse_test.py]: Thanks! Done.

**Comment by tlu7** [comment]
> Let's just consolidate tests, otherwise this looks great!

Thanks! PTAL.

File: tests/sparse_test.py [hunk 2]
```diff
@@ -2548,17 +2548,18 @@ def test_matmul(self, shape, dtype, Obj, bshape):
...
     dtype=jtu.dtypes.floating + jtu.dtypes.complex,
   )
-  def test_bcoo_from_scipy_sparse(self, input_type, shape, dtype):
+  def test_bcoo_bcsr_from_scipy_sparse(self, cls, input_type, shape, dtype):
     rng = rand_sparse(self.rng())
     M = rng(shape, dtype)
     M_sparse = input_type(M)
-    M_bcoo = sparse.BCOO.from_scipy_sparse(M_sparse)
-    self.assertArraysEqual(M, M_bcoo.todense())
+    M_batched = cls.from_scipy_sparse(M_sparse)
```
Comment (jakevdp) [inline tests/sparse_test.py:2561]: minor nitpick, but I'd change `M_sparse` to `M_scipy` and change `M_batched` to `M_jax`

Comment (tlu7) [reply tests/sparse_test.py:2561]: Like the names!

**Review by jakevdp — APPROVED** [review]
Looks great, thanks!

**Pull request merged**